object/slicer/slicer_test.go

```diff
@@ -259,25 +259,43 @@ func (x *slicedObjectChecker) InitDataStream(hdr object.Object) (io.Writer, erro
 
 	x.chainCollector.handleOutgoingObject(hdr, buf)
 
-	return newSizeChecker(x.tb, buf, x.input.payloadLimit), nil
+	return newSizeChecker(x.tb, hdr, buf, x.input.payloadLimit), nil
 }
 
 type writeSizeChecker struct {
-	tb        testing.TB
-	limit     uint64
-	processed uint64
-	base      io.Writer
+	tb         testing.TB
+	hdr        object.Object
+	limit      uint64
+	processed  uint64
+	base       io.Writer
+	payloadWas bool
```

Comment:
Minor. For me bool `wasPayload` looks better then `payloadWas`. But I'm ok left as is

Reply:
`hasPayload`?

Reply:
@smallhive, @roman-khimov, i meant "payload WAS here" OR "payload WAS NOT here". `writeSizeChecker` does not store any payload explicitly, it just can understand wheather it has met any payload or not. if you still think `hasPayload` is a better choice, i dont mind changing it, just ping me again

Reply:
> wheather it has met any payload or not

Well, that's `hasPayload`. It doesn't imply "the payload is _here_" to me. `payloadExists` is another possibility.

Reply:
`payloadSeen` as well.

Reply:
changed to `payloadSeen`